diff --git forkSrcPrefix/src/app/pages/country/country.component.ts forkDstPrefix/src/app/pages/country/country.component.ts
index eae4ef2444f13cfe0b2ce72024bbf862bda300b3..5e6159728f4376592e2b9782e787ff77dd1f0590 100644
--- forkSrcPrefix/src/app/pages/country/country.component.ts
+++ forkDstPrefix/src/app/pages/country/country.component.ts
@@ -1,6 +1,7 @@
 import {HttpClient, HttpErrorResponse} from '@angular/common/http';
-import {Component, OnInit} from '@angular/core';
+import {Component, OnInit, OnDestroy} from '@angular/core';
 import {ActivatedRoute, ParamMap, Router} from '@angular/router';
+import { Country, Participation } from '../../models/olympic.model'
 import Chart from 'chart.js/auto';
 
 
@@ -9,11 +10,11 @@ import Chart from 'chart.js/auto';
   templateUrl: './country.component.html',
   styleUrls: ['./country.component.scss']
 })
-export class CountryComponent implements OnInit {
+export class CountryComponent implements OnInit, OnDestroy {
   private olympicUrl = './assets/mock/olympic.json';
   public lineChart!: Chart<"line", string[], number>;
   public titlePage: string = '';
-  public totalEntries: any = 0;
+  public totalEntries: number = 0;
   public totalMedals: number = 0;
   public totalAthletes: number = 0;
   public error!: string;
@@ -24,17 +25,17 @@ export class CountryComponent implements OnInit {
   ngOnInit() {
     let countryName: string | null = null
     this.route.paramMap.subscribe((param: ParamMap) => countryName = param.get('countryName'));
-    this.http.get<any[]>(this.olympicUrl).pipe().subscribe(
+    this.http.get<Country[]>(this.olympicUrl).pipe().subscribe(
       (data) => {
         if (data && data.length > 0) {
-          const selectedCountry = data.find((i: any) => i.country === countryName);
-          this.titlePage = selectedCountry.country;
-          const participations = selectedCountry?.participations.map((i: any) => i);
+          const selectedCountry = data.find((i: Country) => i.country === countryName);
+          this.titlePage = selectedCountry?.country ?? '';
+          const participations = selectedCountry?.participations.map((i: Participation) => i);
           this.totalEntries = participations?.length ?? 0;
-          const years = selectedCountry?.participations.map((i: any) => i.year) ?? [];
-          const medals = selectedCountry?.participations.map((i: any) => i.medalsCount.toString()) ?? [];
+          const years = selectedCountry?.participations.map((i: Participation) => i.year) ?? [];
+          const medals = selectedCountry?.participations.map((i: Participation) => i.medalsCount.toString()) ?? [];
           this.totalMedals = medals.reduce((accumulator: any, item: any) => accumulator + parseInt(item), 0);
-          const nbAthletes = selectedCountry?.participations.map((i: any) => i.athleteCount.toString()) ?? []
+          const nbAthletes = selectedCountry?.participations.map((i: Participation) => i.athleteCount.toString()) ?? [];
           this.totalAthletes = nbAthletes.reduce((accumulator: any, item: any) => accumulator + parseInt(item), 0);
           this.buildChart(years, medals);
         }
@@ -46,6 +47,9 @@ export class CountryComponent implements OnInit {
   }
 
   buildChart(years: number[], medals: string[]) {
+
+    this.lineChart?.destroy();
+
     const lineChart = new Chart("countryChart", {
       type: 'line',
       data: {
@@ -64,4 +68,8 @@ export class CountryComponent implements OnInit {
     });
     this.lineChart = lineChart;
   }
+
+  ngOnDestroy() {
+    this.lineChart?.destroy();
+  }
 }
diff --git forkSrcPrefix/src/app/pages/home/home.component.ts forkDstPrefix/src/app/pages/home/home.component.ts
index 5b9279b4be6be1659ccdbdd4711de2cc73a5f6eb..6a27f1b71024e40ed17cd4e9272df1e0d37f8a9d 100644
--- forkSrcPrefix/src/app/pages/home/home.component.ts
+++ forkDstPrefix/src/app/pages/home/home.component.ts
@@ -1,5 +1,5 @@
 import {HttpClient, HttpErrorResponse} from '@angular/common/http';
-import {Component, OnInit} from '@angular/core';
+import {Component, OnInit, OnDestroy} from '@angular/core';
 import { Router } from '@angular/router';
 import Chart from 'chart.js/auto';
 
@@ -8,7 +8,7 @@ import Chart from 'chart.js/auto';
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.scss'],
 })
-export class HomeComponent implements OnInit {
+export class HomeComponent implements OnInit, OnDestroy {
   private olympicUrl = './assets/mock/olympic.json';
   public pieChart!: Chart<"pie", number[], string>;
   public totalCountries: number = 0
@@ -39,6 +39,9 @@ export class HomeComponent implements OnInit {
   }
 
   buildPieChart(countries: string[], sumOfAllMedalsYears: number[]) {
+
+    this.pieChart?.destroy();
+
     const pieChart = new Chart("DashboardPieChart", {
       type: 'pie',
       data: {
@@ -66,5 +69,9 @@ export class HomeComponent implements OnInit {
     });
     this.pieChart = pieChart;
   }
+
+   ngOnDestroy() {
+    this.pieChart?.destroy();
+  }
 }
 
